span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/SpanNormalizer.java

```diff
@@ -53,7 +58,13 @@ public StreamsBuilder buildTopology(
 
     KStream<byte[], PreProcessedSpan> preProcessedStream =
         inputStream.transform(JaegerSpanPreProcessor::new);
-    preProcessedStream.transform(JaegerSpanToAvroRawSpanTransformer::new).to(outputTopic);
+
+    KStream<TraceIdentity, RawSpan>[] branches =
+        preProcessedStream
+            .transform(JaegerSpanToAvroRawSpanTransformer::new)
+            .branch(new ByPassPredicate(jobConfig), (key, value) -> true);
+    branches[0].transform(RawSpanToStructuredTraceTransformer::new).to(bypassOutputTopic);
```

Comment:
So braches[0] maps to the case where the predicate is `true`?

Reply:
Here, branch[0] is the bypass branch. The kstream branch API takes an array of predicates, the first predicate that evaluates to true, that branch gets the message. So, here, I have created bypass predicate and default predicate (which is always true). And the index refers to the branch corresponding to the predicate index in an array.

Ref: https://kafka.apache.org/21/javadoc/org/apache/kafka/streams/kstream/KStream.html#branch-org.apache.kafka.streams.kstream.Predicate...-

Reply:
@laxman-traceable Shall I put the counter metric for this? So we can differentiate between bypass vs non-bypass traffic?